assets/scripts/admin/tokens.js

```diff
@@ -208,6 +208,18 @@ function validateAdditionalAttributes() {
     return valid;
 }
 
+function validateEmptyTokenForm() {
+    if ($('#edittoken').find('[name="subaction"]').val() != 'inserttoken') {
+        return true;
+    }
+    var isFormEmpty = $('#email').val() == '' && $('#firstname').val() == '' && $('#lastname').val() == '';
```

Comment:
Need to check attribute too : (for example) client id set.

Reply:
Sorry, don't understand. What is "client id"?
Is that a custom attribute?

We can't check emptyness with attributes which are not always going to be used.

Reply:
attribute_1 for example 

isFormEmpty must check if any value is entered : 

1. can be default one (email, firstname, lastname) : OK
2. attribute_X if exist
3. token 
4. others ? mailsatus (!OK), usesleft (!1)? etc …

1 + 2 + 3 mlust be checked. for 4 : we can leave it like this.

Reply:
This is just a warning for preventing blank records. Not blocking anything.
Checking for token and all possible attribute isn't it going too far?

Reply:
It's not only a warning… it take time … clearly enter anything in attribute or token must not be confirmed.